solidity/test/LPRewardsTest.js

```diff
@@ -0,0 +1,211 @@
+const {contract, web3, accounts} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("./helpers/snapshot")
+const {time} = require("@openzeppelin/test-helpers")
+
+const LPRewards = contract.fromArtifact("LPRewards")
+const KeepToken = contract.fromArtifact("KeepToken")
+const WrappedToken = contract.fromArtifact("TestToken")
+
+const BN = web3.utils.BN
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("LPRewards", () => {
+  const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
+  const allocationForDistribution = web3.utils.toWei("10000000")
+
+  const staker1 = accounts[1]
+  const staker2 = accounts[2]
+  const rewardDistribution = accounts[3]
+  const wrappedTokenOwner = accounts[4]
+  const lpRewardsOwner = accounts[5]
+  const keepTokenOwner = accounts[6]
+
+  let keepToken
+  let lpRewards
+  let wrappedToken
+
+  before(async () => {
+    keepToken = await KeepToken.new({from: keepTokenOwner})
+    // This is a "Pair" Uniswap Token which is created here:
+    // https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2Factory.sol#L23
+    //
+    // There are 3 addresses for the following pairs:
+    // - KEEP/ETH (https://info.uniswap.org/pair/0xe6f19dab7d43317344282f803f8e8d240708174a)
+    // - TBTC/ETH (https://info.uniswap.org/pair/0x854056fd40c1b52037166285b2e54fee774d33f6)
+    // - KEEP/TBTC (https://info.uniswap.org/pair/0x38c8ffee49f286f25d25bad919ff7552e5daf081)
+    wrappedToken = await WrappedToken.new({from: wrappedTokenOwner})
+    lpRewards = await LPRewards.new(keepToken.address, wrappedToken.address, {
+      from: lpRewardsOwner,
+    })
+
+    await keepToken.approve(rewardDistribution, allocationForDistribution, {
+      from: keepTokenOwner,
+    })
+    await keepToken.transfer(rewardDistribution, allocationForDistribution, {
+      from: keepTokenOwner,
+    })
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("tokens allocation", () => {
+    it("should successfully transfer KEEP and notify LPRewards contract on rewards distribution", async () => {
+      const initialBalance = await keepToken.balanceOf(lpRewards.address)
```

Comment:
I'd rather assert it is zero, just like in the next test, or do not care about initial balance at all assuming it's zero.

Reply:
Removed